.github/workflows/tests-runner.yml

```diff
@@ -0,0 +1,195 @@
+          path: |
+            ${{ github.workspace }}/artifacts/log/*/*.binlog
+            ${{ github.workspace }}/artifacts/log/*/TestLogs/**
+            ${{ github.workspace }}/artifacts/tmp/*/combined_runsheet.json
+          retention-days: 3
+
+  generate_e2e_matrix:
+    name: Generate E2E test runsheet
+    runs-on: windows-latest
+    outputs:
+      runsheet: ${{ steps.generate_e2e_matrix.outputs.runsheet }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Build with packages
+        run: |
+          ./build.cmd -restore -build -pack -c Release -ci -bl /p:InstallBrowsersForPlaywright=false /p:SkipTestProjects=true /p:CI=false
+
+      - name: Generate test runsheet
+        id: generate_e2e_matrix
+        run: |
+          ./build.cmd -test /p:TestRunnerName=TestRunsheetBuilder /p:FullE2e=true -bl -c Release
+
+      - name: Upload built NuGets
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: built-nugets
+          path: artifacts/packages
+          retention-days: 3
+
+      - name: Upload logs, and test results
+        if: ${{ always() }}
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: runsheet-e2e-logs
+          path: |
+            ${{ github.workspace }}/artifacts/log/*/*.binlog
+            ${{ github.workspace }}/artifacts/log/*/TestLogs/**
+            ${{ github.workspace }}/artifacts/tmp/*/combined_runsheet.json
+          retention-days: 3
+
+  run_tests:
+    name: Test
+    needs: generate_tests_matrix
+    strategy:
+      fail-fast: false
+      matrix:
+        tests: ${{ fromJson(needs.generate_tests_matrix.outputs.runsheet) }}
+
+    runs-on: ${{ matrix.tests.os }} # Use the OS from the matrix
+
+    steps:
+      - name: Trust HTTPS development certificate
+        if: runner.os == 'Linux'
+        run: dotnet dev-certs https --trust
+
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Verify Docker is running
+        # nested docker containers not supported on windows
+        if: runner.os == 'Linux'
+        run: docker info
+
+      - name: Install Azure Functions Core Tools
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y azure-functions-core-tools-4
+
+      - name: Test ${{ matrix.tests.project }}
+        run: |
+          ${{ matrix.tests.command }}
+
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: ${{ matrix.tests.project }}-${{ matrix.tests.os }}-logs
+          path: |
+            ${{ github.workspace }}/artifacts/TestResults/*/*.trx
+            ${{ github.workspace }}/artifacts/log/*/TestLogs/**
+          retention-days: 3
```

Comment:
Isn't this too short for being able to go back and debug any test failures?

Reply:
This value can be changed as required.